plugins/imagebase/plugin.js

```diff
@@ -614,6 +608,16 @@
 		link: getLinkFeature()
 	};
 
+	function getWidgetsWithFeature( widgets, feature ) {
+		return CKEDITOR.tools.array.reduce( CKEDITOR.tools.objectKeys( widgets ), function( featuredWidgets, id ) {
```

Comment:
Shouldn't it be `filter`, not `reduce`? Then the whole function could be simplified to:

```javascript
return hasWidgetFeature( widgets[ id ], feature );
```

Reply:
I'm not sure how to write it as a filter function? `editor.widgets.instances` is an object so we will have to convert it to an array to use filter function which requires additional code. If I simply replace `reduce` with `filter` here I will get widget ids, not instances. WDYT @Comandeer ?

Reply:
True, I didn't think about it. So let's stay with `reduce` as it's the best option here.